builder/test/simplify/simplify.cc

```diff
@@ -708,6 +708,95 @@ TEST(simplify, slice_of_crop) {
       matches(slice_buffer::make(b3, b0, {{1}, {0}}, body)));
 }
 
+TEST(simplify, slice_of_const_buffer) {
+  stmt body = call_stmt::make(nullptr, {}, {b3}, {});
+  {
+    // Basic test. Slicing a single dimension.
+    slinky::dim dims[2] = {{0, 20, 2}, {0, 0, 42}};
+    raw_buffer_ptr constant_buf = raw_buffer::make(2, 2, dims);
+    std::memset(constant_buf->base, constant_buf->size_bytes(), 0x17);
+
+    raw_buffer_ptr sliced_constant_buf = raw_buffer::make_copy(*constant_buf);
+    sliced_constant_buf->rank = 1;
+
+    ASSERT_THAT(simplify(constant_buffer::make(b1, constant_buf, slice_dim::make(b3, b1, 1, 0, body))),
+        matches(constant_buffer::make(b3, sliced_constant_buf, body)));
+  }
+
+  {
+    // Basic test. Slicing two dimensions.
+    slinky::dim dims[2] = {{0, 10, 1}, {0, 20, 10}};
+    raw_buffer_ptr constant_buf = raw_buffer::make(2, 1, dims);
+
+    raw_buffer_ptr sliced_buf = raw_buffer::make_copy(*constant_buf);
+    sliced_buf->slice(/*dim=*/1, /*at=*/10);
+    sliced_buf->slice(/*dim=*/0, /*at=*/5);
+
+    ASSERT_THAT(simplify(constant_buffer::make(b1, constant_buf, slice_buffer::make(b3, b1, {5, 10}, body))),
+        matches(constant_buffer::make(b3, sliced_buf, body)));
+  }
+
+  {
+    // Test out-of-bounds slicing.
+    slinky::dim dims[2] = {{0, 10, 1}, {0, 20, 1}};
+    raw_buffer_ptr constant_buf = raw_buffer::make(2, 1, dims);
+
+    raw_buffer_ptr sliced_buf = raw_buffer::make_copy(*constant_buf);
+    sliced_buf->slice(/*dim=*/1, /*at=*/21);
+    sliced_buf->slice(/*dim=*/0, /*at=*/5);
+    ASSERT_EQ(sliced_buf->base, nullptr);
+
+    ASSERT_THAT(simplify(constant_buffer::make(b1, constant_buf, slice_buffer::make(b3, b1, {5, 21}, body))),
+        matches(constant_buffer::make(b3, sliced_buf, body)));
+  }
+  {
+    // Slicing of a tensor of rank 3.
+    slinky::dim dims[3] = {{0, 1, 1}, {0, 1, 2}, {0, 1, 4}};
+    for (int d = 0; d < 3; ++d) {
```

Comment:
I think this should be 3 separate loops over d. I think this might make it easier to understand which case is failing, (if one is failing) and then you also don't need the dummy blocks.